platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/StaticTopology.java

```diff
@@ -121,4 +115,15 @@ public boolean shouldConnectTo(final NodeId nodeId) {
     public RandomGraph getConnectionGraph() {
         return connectionGraph;
     }
+
+    /**
+     * Returns the index of the given node in the peer list, which must not be the self index
+     *
+     * @param nodeId the node ID
+     * @return the index of the node in the peer list
+     */
+    private int getIndexOfNodeId(@NonNull final NodeId nodeId) {
```

Comment:
Won't this method have an off by 1 error for every node with index > self index? You're using position in `peerNodes`, but self is missing.

Reply:
for every node with index > self index, we need the index returned by this method to be off by 1 (to match their position similar to the original addressBook). 
There's an edge case for when being off by 1 can lead to this method returning an index number > vertex upper bound, but we deal with that correctly in the isAdjacent check so it's ok.

Reply:
I don't understand this either. The previous code in this class did the following, to get the index of the other node in the address book. This computation *includes* self id in the indices

```
addressBook.getIndexOfNodeId(nodeId);
```

But the new code instead defines the index of the other node via its position in the peer list, which *doesn't* contain self id. So for every node with index greater than this self id, the returned node index will be 1 less.

Reply:
Ok, I think I understand why this works.

If the node ID of the neighbor is greater than ours, then this method returns a value that is one less than its actual index. If the node ID of the neighbor is equal to our node ID plus one, getIndexOfNodeId() increments the number so that it doesn't actually equal our index.

This class still returns the correct results because we only care if the other node index is greater or less than our own index, so it doesn't matter that it's off by 1 as long as it is greater than our index.

Although this technically yields the correct result, I feel like this is one of those times where the code is "accidentally returning the right results". At the very least, the code here does not return values consistent with the javadocs. Better to refactor this code I think. I can imagine that the reason why/how this works may not be apparent to a developer who looks at this in the future.